### PR TITLE
chore: add commit-msg hook

### PR DESCRIPTION
**Pull request by chanduvardhan007**
validates the conventional commit message format using husky #8685zwd7h

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,7 +21,13 @@
     "eslint": "^8.45.0",
     "eslint-plugin-react-hooks": "^4.6.0",
     "eslint-plugin-react-refresh": "^0.4.3",
+    "husky": "^8.0.3",
     "typescript": "^5.0.2",
     "vite": "^4.4.5"
+  },
+  "husky": {
+    "hooks": {
+      "commit-msg": "chmod +x ./husky/commit-msg && bash ./husky/commit-msg"
```
Comment (RushilBasappa) [inline package.json]: Setting permissions is a one time command. And file permissions are also saved in git.

Remove chmod from here, and run it on the file and make a commit

File: .husky/commit-msg [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+# if [[ "$commit_message" == '' ]];then
+#     echo "msg is empty"
+#     exit 1
+# fi
+
+# Define the regex for the Conventional Commit format
+
+commit_regex='^(build|ci|docs|feat|fix|perf|refactor|style|test|chore)(\(.+\))?: .{1,50}(\n\n.+)?'
+
```
Comment (RushilBasappa) [inline .husky/commit-msg:11]: This will only check the first line

Is there a way to also check 
1. if ticket id is added
2. Body is written with * etc?

**Review by RushilBasappa — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (chanduvardhan007) [reply package.json]: ok

File: .husky/commit-msg (hunk 2 above)
Comment (chanduvardhan007) [reply .husky/commit-msg:11]: will check and update

**Comment by AbhiGattineni** [comment]
@chanduvardhan007 

1) don't merge main into your branch, this will create another branch( instead use rebase).
2) squash all the commits into 1 commit and force push it(force push is only when you pushed your branch already, try to use force push as many less times as possible).

**Pull request merged**